### PR TITLE
Record inflight metrics during benchmarks

### PR DESCRIPTION
**Pull request by seglo**
## Purpose

Poll JVM, Kafka consumer, and Kafka broker metrics during a benchmark test and write the results to two CSV files:

1. Summary with just last poll of metrics
2. Detail that show inflight metrics each second of test

This PR will be used for tests in the Jenkins benchmark build.  It will also be used to create source data for Kafka 2.4.0 prefetch blog post diagram artifacts.

## References

* Will supersede #837 when complete

## Changes

* Add new Committable and Plain consumer benchmarks for topics with 100 partitions
* Add Plain consumer benchmark that records inflight metrics
* `InflightMetrics` trait consolidates polling of metrics from JVM, consumer, and broker
* Maintain our own testcontainers `KafkaContainer` so we can modify it freely.  Enabled JMX metrics.  Created upstream PR for consideration: https://github.com/testcontainers/testcontainers-java/pull/2192

Misc.

* Use default commit `parallelism` in `BatchedConsumer.consumerAtLeastOnceBatched`
* Remove `sbt-docker` from project

This PR replaces: https://github.com/akka/alpakka-kafka/pull/1004

### CODE REVIEW

File: testkit/src/main/scala/akka/kafka/testkit/KafkaTestkitTestcontainersSettings.scala [hunk 1]
```diff
@@ -8,17 +8,18 @@ package akka.kafka.testkit
...
     val numBrokers: Int,
     val internalTopicsReplicationFactor: Int,
-    val configureKafka: Vector[KafkaContainer] => Unit = _ => (),
-    val configureKafkaConsumer: java.util.function.Consumer[java.util.Collection[KafkaContainer]] =
-      new Consumer[java.util.Collection[KafkaContainer]]() {
-        override def accept(arg: java.util.Collection[KafkaContainer]): Unit = ()
+    val configureKafka: Vector[AlpakkaKafkaContainer] => Unit = _ => (),
+    val configureKafkaConsumer: java.util.function.Consumer[java.util.Collection[AlpakkaKafkaContainer]] =
+      new Consumer[java.util.Collection[AlpakkaKafkaContainer]]() {
+        override def accept(arg: java.util.Collection[AlpakkaKafkaContainer]): Unit = ()
```
Comment (ennru) [inline testkit/src/main/scala/akka/kafka/testkit/KafkaTestkitTestcontainersSettings.scala]: This is public API exposing internal classes. Can these be marked as `@InternalApi`?

File: benchmarks/src/main/scala/akka/kafka/benchmarks/InflightMetrics.scala [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+              val nextInterval = interval + timeMs
+              val nextAcc = accLastMetrics match {
+                case None => Counter.reset(newMetrics)
+                case Some(lastMetrics) => Counter.update(lastMetrics, newMetrics)
+              }
+              (nextInterval, Some(nextAcc))
+            case _ => throw new IllegalStateException("The wrong number of Future results were returned.")
+          }
+      })
+      .mapConcat { case (_, results: Option[List[Metric]]) => results.map(_.map(_.value.toString)).toList }
```
Comment (ennru) [inline benchmarks/src/main/scala/akka/kafka/benchmarks/InflightMetrics.scala]: Great showcase for Akka Streams.

**Comment by bsideup** [comment]
> Maintain our own testcontainers KafkaContainer so we can modify it freely. Enabled JMX metrics. Created upstream PR for consideration: testcontainers/testcontainers-java#2192

Please be aware of this comment:
https://github.com/testcontainers/testcontainers-java/pull/2192#issuecomment-567909070

it seems to be a bit better approach, that also works on Mac & Windows

File: testkit/src/main/scala/akka/kafka/testkit/KafkaTestkitTestcontainersSettings.scala (hunk 1 above)
Comment (seglo) [reply testkit/src/main/scala/akka/kafka/testkit/KafkaTestkitTestcontainersSettings.scala]: On `AlpakkaKafkaContainer`?

**Pull request merged**